metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -303,14 +303,23 @@ object MetalsEnrichments
 
   implicit class XtensionAbsolutePathBuffers(path: AbsolutePath) {
 
-    def hasScalaFiles: Boolean = {
-      def isScalaDir(file: File): Boolean = {
+    def isScalaProject(extendSearchToScriptsAndSbt: Boolean = true): Boolean = {
+      val isScala: String => Boolean =
+        if (extendSearchToScriptsAndSbt) _.isScalaFilename else _.isScala
```

Comment:
What about Java source files, since Metals does provide some useful support for it? :slightly_smiling_face:

Reply:
Metals isn't really a Java LSP and for a pure Java project one should most probably use something else. However, you can always force metals (server) to acknowledge smth as a "metals project" by adding `.metals` dir.

Reply:
Yeah, ideally you would start Java LSP server for those workspaces that are pure Java